src/components/dialog/Dialog.js

```diff
@@ -1,102 +1,76 @@
 import React, { PureComponent } from 'react';
-import { createPortal } from 'react-dom';
 import PropTypes from 'prop-types';
-import cx from 'classnames';
-import Overlay from '../overlay/Overlay';
-import Transition from 'react-transition-group/Transition';
-import theme from './theme.css';
+import omit from 'lodash.omit';
+
+import { DialogBase } from './index';
+import { Banner, Box, Button, ButtonGroup, COLORS, Heading2, Heading3, Link } from '../../index';
 
 class Dialog extends PureComponent {
-  dialogRoot = document.createElement('div');
+  getHeader = () => {
+    const { headerColor, headerIcon, headingLevel, onCloseClick, title } = this.props;
 
-  componentDidMount() {
-    document.body.appendChild(this.dialogRoot);
-  }
+    return (
+      <Banner color={headerColor} fullWidth icon={headerIcon} onClose={onCloseClick}>
+        {headingLevel === 2 ? <Heading2>{title}</Heading2> : <Heading3>{title}</Heading3>}
+      </Banner>
+    );
+  };
 
-  componentWillUnmount() {
-    document.body.removeChild(this.dialogRoot);
-  }
+  getFooter = () => {
+    const { tertiaryAction, secondaryAction, primaryAction } = this.props;
 
-  render() {
-    const {
-      active,
-      backdrop,
-      children,
-      className,
-      onEscKeyDown,
-      onOverlayClick,
-      onOverlayMouseDown,
-      onOverlayMouseMove,
-      onOverlayMouseUp,
-      size,
-    } = this.props;
+    return (
+      <ButtonGroup justifyContent="flex-end" padding={4}>
+        {tertiaryAction && <Link inherit={false} {...tertiaryAction} />}
```

Comment:
Q: when are we supposed to use `<Link />` and when are we supposed to use `<Button level="link" />`?

Reply:
Using the `<Button level="link" />` does not visually match well next to the other buttons. Due to its transparent background and padding you get the feeling the buttons are not equally spaced. This is not the case with the `Link`.

@stephen-tl thinks that the `<Button level="link" />` only is suited for specific cases, such as the pagination for example.